leaf-ui/rappid-extensions/LinkInspector.js

```diff
@@ -28,103 +33,177 @@ var LinkInspector = Backbone.View.extend({
+                '<option value="+">+</option>',
+                '<option value="-">-</option>',
+                '<option value="+S">+S</option>',
+                '<option value="++S">++S</option>',
+                '<option value="-S">-S</option>',
+                '<option value="--S">--S</option>',
+                '<option value="+D">+D</option>',
+                '<option value="++D">++D</option>',
+                '<option value="-D">-D</option>',
+                '<option value="--D">--D</option>',
             '</select>',
-            '<select id="link-type-end" class="repeat-select">',
+            '<select id="link-type-end" class="repeat-select" <% if (!selected) {%> disabled= true style = "background-color:gray"<%} %>>',
+                '<option value="NO">No Relationship</option>',
+                '<option value="AND" class = "A">And-Decomposition</option>', //class works with function in line 294 and 296
+                '<option value="OR" class = "A">Or-Decomposition</option>',
+                '<option value="++" class = "B">++</option>',
+                '<option value="--" class = "B">--</option>',
+                '<option value="+" class = "B">+</option>',
+                '<option value="-" class = "B">-</option>',
+                '<option value="+S" class = "B">+S</option>',
+                '<option value="++S" class = "B">++S</option>',
+                '<option value="-S" class = "B">-S</option>',
+                '<option value="--S" class = "B">--S</option>',
+                '<option value="+D" class = "B">+D</option>',
+                '<option value="++D" class = "B">++D</option>',
+                '<option value="-D" class = "B">-D</option>',
+                '<option value="--D" class = "B">--D</option>',
             '</select>',
             '<button id="switch-link-type" class="inspector-btn small-btn blue-btn">Constant Relationships</button>',
-            '<br>'
+            '<br>',
+            '</div>',
+            '</script>'
     ].join(''),
 
     actortemplate: [
+        '<script type="text/template" id="item-template">',
             '<label> Link Type </label> <br>',
+            '<div class="inspector-views">',
             '<select id="actor-link" class="link-type">',
                 '<option value="is-a">is-a</option>',
                 '<option value="plays">plays</option>',
                 '<option value="is-part-of">is-part-of</option>',
-            '</select><br>'].join(''),
+            '</select><br>',
+            '</div>',
+            '</script>'
+        ].join(''),
+
+    errortemplate: [
+        '<script type="text/template" id="item-template">',
+            '<label> Error </label> <br>',
+            '<div class="inspector-views">',
+            '</div>',
+            '</script>'
+        ].join(''),
 
     events: {
-            'click #switch-link-type': 'switchMode',
+            'click #switch-link-type': 'renderNewMode',
```

Comment:
What if you put a separate id for the switch button in the evolving template vs the constant template here instead? That way each id could call a different function that either switched to constant mode or to evolving mode depending. This would simplify a lot of your logic